src/main/scala/beam/router/model/BeamLeg.scala

```diff
@@ -126,20 +126,20 @@ object BeamLeg {
   }
 
   def makeVectorLegsConsistentAsOrderdStandAloneLegs(legs: Vector[BeamLeg]): Vector[BeamLeg] = {
-    if (legs.isEmpty) {
-      legs
-    } else {
-      var latestEndTime = legs.head.startTime - 1
-      var newLeg = legs.head
-      for (leg <- legs) yield {
-        if (leg.startTime < latestEndTime) {
-          newLeg = leg.updateStartTime(latestEndTime)
-        } else {
-          newLeg = leg
+    legs.headOption match {
+      case None => legs
+      case Some(legHead) =>
```

Comment:
Do we need to care it is the head - if so then I'd say we should go with headLeg or topLeg, firstLeg

Reply:
variable renamed